clarity/src/proptesting/values.rs

```diff
@@ -0,0 +1,255 @@
+use proptest::prelude::*;
+
+use super::*;
+use crate::vm::types::{
+    BuffData, CharType, ListData, ListTypeData, OptionalData, PrincipalData,
+    QualifiedContractIdentifier, ResponseData, SequenceData, SequenceSubtype,
+    StandardPrincipalData, StringSubtype, TupleData, TupleTypeSignature, TypeSignature, UTF8Data,
+};
+use crate::vm::{ContractName, Value};
+
+/// Returns a [`Strategy`] for generating a randomized [`Value`] instance of a
+/// the specified ([`TypeSignature`]).
+pub fn value(ty: TypeSignature) -> impl Strategy<Value = Value> {
+    match ty {
+        TypeSignature::NoType => unreachable!(),
+        TypeSignature::IntType => int().boxed(),
+        TypeSignature::UIntType => uint().boxed(),
+        TypeSignature::BoolType => bool().boxed(),
+        TypeSignature::OptionalType(ty) => optional(*ty).boxed(),
+        TypeSignature::ResponseType(ok_err) => response(ok_err.0, ok_err.1).boxed(),
+        TypeSignature::SequenceType(SequenceSubtype::BufferType(size)) => {
+            buffer(size.into()).boxed()
+        }
+        TypeSignature::SequenceType(SequenceSubtype::StringType(StringSubtype::ASCII(size))) => {
+            string_ascii(size.into()).boxed()
+        }
+        TypeSignature::SequenceType(SequenceSubtype::StringType(StringSubtype::UTF8(size))) => {
+            string_utf8(size.into()).boxed()
+        }
+        TypeSignature::SequenceType(SequenceSubtype::ListType(list_type_data)) => {
+            list(list_type_data).boxed()
+        }
+        TypeSignature::TupleType(tuple_ty) => tuple(tuple_ty).boxed(),
+        TypeSignature::PrincipalType => {
+            prop_oneof![principal_standard(), principal_contract()].boxed()
+        }
+        // TODO
+        TypeSignature::ListUnionType(_) => todo!(),
```

Comment:
This is not really a TODO anymore: `ListUnionType` are created by accident when a List of qualified principal is generated.

Reply:
I'm not sure that I want to do anything here just yet as it's not so clear how this will evolve within `clarity` as more property tests are introduced, and it's nothing specifically required by this PR. OK to resolve?